04.bowling/bowling.rb

```diff
@@ -0,0 +1,66 @@
+def main
+  point = get_point(ARGV[0])
+  puts point
+end
+
+def get_point(score_str)
+  shots = get_shots(score_str)
+  frames = get_frames(shots)
+  calc_total_point(frames)
+end
+
+def get_shots(score_str)
+  scores = score_str.split(',')
+  scores.flat_map do |s|
+    s == 'X' ? [10, 0] : s.to_i
+  end
+end
+
+def get_frames(shots)
+  frames = shots.each_slice(2).to_a
+
+  if frames.length > 10
+    final_frame = frames[9..].flat_map do |frame|
+      frame[0] == 10 ? 10 : frame
+    end
+    frames = [*frames[..8], final_frame]
+  end
+  frames
+end
+
+def calc_total_point(frames)
+  point = 0
+  (0..8).each do |i|
+    current_frame = frames[i]
+    next_frame = frames[i + 1]
+    after_next_frame = frames[i + 2]
+    is_one_before_final_frame = (i == 8)
+
+    point += calc_frame_point(current_frame, next_frame, after_next_frame, is_one_before_final_frame)
```

Comment:
```suggestion
    point += calc_frame_point(frames, i)
```

のようにした方が引数が減るのでベターかもしれません。

Reply:
calc_frame_point(frames, i)にすると`calc_frame_point `をメソッドに切り出す必要がなくなると思い、`calc_total_point`の中にループ処理もを書くようにしました。
https://github.com/mayumonj/ruby-practices/commit/1637707c5fc2fa6b46b9225c160787d882e952cb